sklearn/utils/testing.py

```diff
@@ -882,3 +903,161 @@ def check_docstring_parameters(func, doc=None, ignore=None, class_name=None):
+        The collection (list, set etc.) of objects that may be either
+        ``NumpyDocString`` instances or objects (classes, functions,
+        descriptors) with docstrings that can be parsed by numpydoc.
+
+    include_params : collection, False or True (default)
+        Collection of Parameters to be included. True, for including all
+        parameters.
+
+    exclude_params : collection or None (default)
+        Collection of Parameters to be excluded. Set only if
+        ``include_params`` is True.
+
+    include_attribs : collection, False or True (default)
+        Collection of Attributes to be included. True, for including all
+        attributes.
+
+    exclude_attribs : collection or None (default)
+        Collection of Attributes to be excluded. Set only if
+        ``include_attribs`` is True.
+
+    include_returns : collection, False or True (default)
+        Collection of Returns to be included. True, for including all returns.
+
+    exclude_returns : collection or None (default)
+        Collection of Returns to be excluded. Set only if ``include_returns``
+        is True.
+
+    Notes
+    -----
+    This function asserts that any Parameters/Returns/Attributes entries having
+    the same name among ``objects`` docstrings also have the same type
+    specification and description.
+    It compares only those entries having the same name in the docstring, does
+    no comparison if an entry in a docstring is unique.
+
+    Examples
+    --------
+    >>> from sklearn.metrics import (mean_absolute_error, mean_squared_error,
+    ... median_absolute_error)
+    >>> from sklearn.utils.testing import assert_consistent_docs
+    ... # doctest: +SKIP
+    >>> assert_consistent_docs([mean_absolute_error, mean_squared_error],
+    ... include_params=['y_true', 'y_pred', 'sample_weight'],
+    ... include_attribs=False, include_returns=False)  # doctest: +SKIP
+    >>> assert_consistent_docs([median_absolute_error, mean_squared_error],
+    ... include_params=True, include_attribs=False, include_returns=False)
+    ... # doctest: +NORMALIZE_WHITESPACE, +SKIP
+    Traceback (most recent call last):
+        ...
+    AssertionError: Parameter 'y_true' of 'mean_squared_error' has inconsistent
+    type definition with that of 'median_absolute_error'.
+
+    """
+    if ((exclude_params and include_params is not True) or
+       (exclude_attribs and include_attribs is not True) or
+       (exclude_returns and include_returns is not True)):
+        raise TypeError("exclude_ argument can be set only if include_"
+                        " argument is True.")
+
+    from numpydoc import docscrape
+
+    # Dictionary of all different Parameters/Attributes/Returns found
+    param_dict = {}
+    attrib_dict = {}
+    return_dict = {}
+
+    i = 1  # sequence of object in the collection
+    for u in objects:
+        if isinstance(u, docscrape.NumpyDocString):
+            doc = u
+            name = 'Object '+str(i)
```

Comment:
space around +. I think we should allow the user to pass in names somehow...

Reply:
Perhaps objects can be (name, numpydocstring) pairs

Reply:
That would be appropriate for numpydocstring objects.
So, now `objects` can be a callable (function,class etc.) or tuple of type `(string, NumpyDocString)`. Am I right?